src/libasr/pass/instantiate_template.cpp

```diff
@@ -191,28 +227,26 @@ class FunctionInstantiator : public ASR::BaseExprStmtDuplicator<FunctionInstanti
                     ASR::Real_t* tnew = ASR::down_cast<ASR::Real_t>(t);
                     return ASRUtils::TYPE(ASR::make_Real_t(al, t->base.loc,
                             tnew->m_kind, param->m_dims, param->n_dims));
-                }  
+                }                  
```

Comment:
You should configure your editor to remove white space, then there is no issue. I am going to merge it, and you can fix it later.